pkg/policy/api/l4.go

```diff
@@ -49,15 +49,17 @@ type PortProtocol struct {
 	// +kubebuilder:validation:Optional
 	EndPort int32 `json:"endPort,omitempty"`
 
-	// Protocol is the L4 protocol. If omitted or empty, any protocol
-	// matches. Accepted values: "TCP", "UDP", "SCTP", "ANY"
+	// Protocol is the L4 protocol. If "ANY", omitted or empty, any protocols
+	// with transport ports (TCP, UDP, SCTP) match.
```

Comment:
Shouldn't ANY also match VRRP and IGMP?

Reply:
The policy engine adds explicit entries only for TCP, UDP, and SCTP when the protocol is set (or defaults) to `ANY` - https://github.com/cilium/cilium/blob/main/pkg/policy/rule.go#L613-L630. 

This PR doesn't change the semantics for ANY. See more context in one of my earlier comments - https://github.com/cilium/cilium/pull/39872#issuecomment-2970758191.

Reply:
I believe this note is descriptive enough - 

```
If "ANY", omitted or empty, any protocols
	// with transport ports (TCP, UDP, SCTP) match.
```

I'm not sure if it's worth mentioning every IP protocol in case we extend the list for other protocols in future. Happy to extend the note if users report confusion around the `ANY` usage.